### ISSUE TITLE
MAINT: Regular aarch64 and arm64 testing

### ISSUE THREAD

**Issue by larsoner** [scipy/scipy#17119]
> My previous comment was solely directed towards wheel building. It would also be possible to do regular native CI testing on linux_aarch64 and macosx_arm64 (possibly other archs), which we're not currently doing

We probably should do that as well, otherwise we are going to find out at release time that we have issues. Can be done later/separately though.

_Originally posted by @rgommers in https://github.com/scipy/scipy/issues/17029#issuecomment-1252462496_

**Comment by larsoner** [comment]
cc @rgommers @andyfaff @tylerjereddy @tupui since you were involved in #17029, I wanted to create a new issue to discuss the non-wheel-building aspects of this, since it seems like it should be a follow-up PR after #17029.

> We probably should do [regular native CI testing] as well, otherwise we are going to find out at release time that we have issues. Can be done later/separately though.

I recently implemented Cirrus for macOS-arm64 [on another repo](https://github.com/mne-tools/mne-python/blob/main/.cirrus.yml) and it was fairly painless, but it seems like the run concurrency is 1 job -- so if there are multiple PRs things can get backed up. However, it looks like #17029 avoids this so far by only triggering jobs on specific commit messages, which is great! :+1: 

Also it seems like the run speed for the macOS-arm64 was quite good -- speed on par with the native Linux builds, and seemingly faster than the macOS-Intel GH actions ones.

In any case, we should be a bit careful about this in future PRs that make Cirrus more widely used to make sure we don't create a CI job backlog. Maybe a scheduled job once a day or so with emails sent to some maintainers on failure, that way we can at least play catch up...?

**Comment by rgommers** [comment]
> Maybe a scheduled job once a day or so with emails sent to some maintainers on failure, that way we can at least play catch up...?

That sounds like a good idea to me. The vast majority of PRs are pure Python, docs or other things that aren't going to behave differently on macOS. So scheduled runs are probably enough.

**Comment by andyfaff** [comment]
When the cibuildwheel related infrastructure is completed (#17029 and #17086 are merged) we will be building wheels on a weekly basis for all targets. Part of the build process is to run the test suite, so we will be doing comprehensive scheduled tests once a week. We could adjust the cron to daily.

However, it would be really simple to adjust the cirrus-ci starlark script to do a single `cp3x-macosx_arm64` and a single `cp3x-linux_aarch64` run as part of a 'regular/normal' CI test. Each of those runs would be on the order of 15-20 mins of CI time.

**Comment by andyfaff** [comment]
cirrus-ci doesn't send email notifications by itself, but points to this separate github action: https://cirrus-ci.org/guide/notifications/

**Comment by rgommers** [comment]
> However, it would be really simple to adjust the cirrus-ci starlark script to do a single `cp3x-macosx_arm64` and a single `cp3x-linux_aarch64` run as part of a 'regular/normal' CI test. Each of those runs would be on the order of 15-20 mins of CI time.

This sounds like a great idea to me.
